py/dml/codegen.py

```diff
@@ -1394,6 +1394,21 @@ def stmt_compound(stmt, location, scope):
     [stmt_asts] = stmt.args
     lscope = Symtab(scope)
     statements = codegen_statements(stmt_asts, location, lscope)
+    if (logging.show_porting and stmt.site.dml_version() == (1, 2)
+        and not stmt.site.filename().endswith('dml-builtins.dml')):
+        method = location.method()
+        if method is not None and len(method.outp) == 1:
```

Comment:
Why restrict this check to the single-output case? It could be generalized -- go through the list, once you encounter an assignment to an output parameter, walk through the subsequent statements as long as they're assignments to output parameters. Upon the first non-output parameter assignment you encounter, if it's a return, report `POUTARGRETURN` for all the output parameters you've encountered.

Reply:
Also, note that the typical two-outarg case is something like:
```
method m() -> (bool ok, int value) {
  if ($valid()) {
   ok = false;
   return;
 } else {
   ok = true;
   value = $result();
 }
}
```
where one of the args is dontcare, which is somewhat hard to translate automatically.

Reply:
I realized that I missed one case: I don't remove the intermediate variable in the case `call $foo() -> (value); return;`  where `value` is an outarg. Crude grepping suggests that this happens in about 10 cases out of ~300 in our repo, so not necessarily worthwhile fixing.

Reply:
Hmm! That's an argument in favor of checking the `ctree` representation! DML 1.2 calls with one return value are codegen:d using `mkCopyData`, i.e. `ExpressionStatement(AssignOp(...))`, in order to assign to the outarg, which means the old check _would_ catch that case.

Reply:
It would, but the porting rule converts `x = Y;` by replacing the `x =` string with `return `, which may be problematic if the sites point into the expression `call Y() -> (x)`. Though, with a bit of luck the porting script might have converted that call to `x = Y()` and managed to do a corresponding transformation of all sites.

Reply:
Up to you. Like you said, it's about a "good enough" heuristic. I'm ok with approving this PR as-is.